File: CPP/explore/Rotate.cpp
#include <vector>

class Solution
{
public:
  void rotate(std::vector<int> &nums, int k)
  {
    // No rotation.
    if (k % nums.size() == 0)
      return;

    std::vector<int> numsCopy(nums.size());
    for (int i = 0; i < nums.size(); i++)
    {
      numsCopy[i] = nums[i];
    }

    for (int i = 0; i < nums.size(); i++)
    {
      nums[(i + k) % nums.size()] = numsCopy[i];
    }
  }
};

File: JS/explore/easy/Rotate.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var rotate = function(nums, k) {
	// Copy by value
	let numsCopy = [ ...nums ];

	for (let i = 0; i < nums.length; i++) {
		nums[(i + k) % nums.length] = numsCopy[i];
		console.log('Value at index', (i + k) % nums.length, 'is now', numsCopy[i]);
	}
};
